torch/distributed/_tensor/placement_types.py

```diff
@@ -184,11 +184,11 @@ def _reduce_shard_tensor(
         """
         my_coordinate = mesh.get_coordinate()
         num_chunks = mesh.size(dim=mesh_dim)
-        # TODO: what should happen if rank is not in the mesh?
-        # see issue https://github.com/pytorch/tau/pull/492
-        assert (
-            my_coordinate is not None
-        ), "Rank if not part of mesh"  # TODO: figure out behavior here
+
+        if my_coordinate is None:
+            # if rank is not part of mesh, we simply return local_tensor,
+            # which should be an empty tensor
+            return tensor
```

Comment:
So this is essentially a no-op right?

Reply:
yes, it's a no-op if rank is not part of mesh